member/src/main/scala/com/improving/app/member/api/MemberServiceImpl.scala

```diff
@@ -0,0 +1,128 @@
+package com.improving.app.member.api
+import akka.actor.typed.ActorSystem
+import akka.cluster.sharding.typed.scaladsl.{ClusterSharding, Entity}
+import akka.cluster.typed.{Cluster, Join}
+import akka.pattern.StatusReply
+import akka.serialization.jackson.JacksonObjectMapperProvider
+import akka.util.Timeout
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.improving.app.member.domain.Member.{MemberCommand, MemberEntityKey}
+import com.improving.app.member.domain._
+import com.improving.app.member.utils.serialize.AvroJacksonObjectMapperFactory.applyObjectMapperMixins
+import wvlet.airframe.ulid.ULID
+
+import scala.concurrent.duration.DurationInt
+import scala.concurrent.{ExecutionContext, Future}
+import scala.language.postfixOps
+
+class MemberServiceImpl(implicit val system: ActorSystem[_]) extends MemberService {
+
+  implicit val ec: ExecutionContext = system.executionContext
+  implicit val timeout: Timeout = Timeout(5 seconds)
+
+  //Create a new member
+  val sharding: ClusterSharding = ClusterSharding(system)
+  ClusterSharding(system).init(
+    Entity(MemberEntityKey)(entityContext => Member(entityContext.entityTypeKey.name, entityContext.entityId))
+  )
+
+  Cluster(system).manager ! Join(Cluster(system).selfMember.address)
+
+  implicit val objectMapper: ObjectMapper = JacksonObjectMapperProvider(system).getOrCreate("jackson-cbor", None)
+  applyObjectMapperMixins(objectMapper)
+
+  //Do not use for RegisterMember
+  private def extractEntityId(request: MemberRequest): String = {
+    request match {
+      case RegisterMember(_, _)             => None
+      case ActivateMember(memberId, _)      => memberId.map(_.id)
+      case InactivateMember(memberId, _)    => memberId.map(_.id)
+      case SuspendMember(memberId, _)       => memberId.map(_.id)
+      case TerminateMember(memberId, _)     => memberId.map(_.id)
+      case UpdateMemberInfo(memberId, _, _) => memberId.map(_.id)
+      case GetMemberInfo(memberId)          => memberId.map(_.id)
+      case other                            => throw new RuntimeException(s"Unexpected request to extract id $other")
+    }
+  }.getOrElse(throw new RuntimeException(s"Missing member id in request $request"))
+
+  private def handleResponse[T](
+      eventHandler: PartialFunction[StatusReply[MemberResponse], T]
+  ): PartialFunction[StatusReply[MemberResponse], T] = {
+    eventHandler.orElse({
+      case StatusReply.Success(response) => throw new RuntimeException(s"Unexpected response $response")
```

Comment:
I am merging multiple partial functions here - the idea is a follows in case of RegisterMember request - we expect a MemberRegistered response only
 So we supply the PF in def registerMember(in: RegisterMember)
      { case StatusReply.Success(MemberEventResponse(response @ MemberRegistered(_, _, _, _))) =>
        response
      }

then the other part gets attached to it in the common def handleResponse[T] function


    eventHandler.orElse({
      case StatusReply.Success(response) => throw new RuntimeException(s"Unexpected response $response")
      case StatusReply.Error(ex)         => throw ex
    })

so it becomes collectively

    {
      case StatusReply.Success(MemberEventResponse(response @ MemberRegistered(_, _, _, _))) =>
        response
      case StatusReply.Success(response) => throw new RuntimeException(s"Unexpected response $response")
      case StatusReply.Error(ex)         => throw ex
    }

I just did this to avoid code repetition

Reply:
Okay, I missed that it was a PF.